src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -129,8 +133,8 @@ private void CheckForShutdown()
 
             if (_connection != null)
             {
+                WebtransportManager?.Dispose();
```

Comment:
Is WebTransport session stream counted towards `_activeRequests` collection in `Http3Connection?

Reply:
Yes it is

Reply:
when is it removed from there?

Reply:
I mean, in normal HTTP/3 request, the stream will be removed from the collection when the request is finished. In our case, does it stay inside the collection until the user closes the session, or is it removed earlier?

Reply:
I will add it in the disposeAsync function since we keep the http3 connection and I can call that function

Reply:
Did not see your last comment, updating
I think that is a delicate question. The request should stay active until the user disposes the session. But i cannot say it is an "active" request anylonger since we do not do anything with it and we dispose. But I think it is best to stay inside the _activeRequests collection until the session is done. But I realised when I dispose it that I do not remove the stream from the collection and will update the code to reflect that

Reply:
I also think it should stay inside activeRequests until the user disposes the session. Otherwise, the connection might get collected and disposed automatically, which we don't want, as the user might still be using the session. Connections in the connection pool are being recycled automatically from time to time, and activeRequests helps to prevent connection being closed while there are still requests in flight. I think session also counts as the "request in flight" and keep connection alive. But it is correct that it should be removed as soon as the session is disposed.

Reply:
I kept them active until disposing of the session